### PR TITLE
Refactor RzTable: change the implementation to RzVector

### PR DESCRIPTION
**Pull request by PeiweiHu**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Highlight the following changes, they are different from the last pr and need to review. 

change table.c:180 to fix memory leak
change table.c:348 to fix memory leak
change table.c:686 to fix warning
change table.c:766 to fix memory leak
change table.c:865 to fix memory leak
change table.c:991 to fix memory leak

...

**Test plan**

...

**Closing issues**

partially address https://github.com/rizinorg/rizin/issues/1961

...

### CODE REVIEW

File: librz/core/cmd/cmd_analysis.c [hunk 1]
```diff
@@ -7331,13 +7331,13 @@ static void print_stats(RzCore *core, HtPU *ht, RzAnalysisFunction *fcn, RzCmdSt
 		rz_list_foreach (list, iter, name) {
 			rz_table_add_column(t, typeNumber, name, 0);
 		}
-		RzList *items = rz_list_newf(free);
-		rz_list_append(items, strdup(fcn->name));
+		RzPVector *items = rz_pvector_new(free);
```
Comment (XVilka) [inline librz/core/cmd/cmd_analysis.c:7334]: Add NULL check

File: librz/core/cmd/cmd_analysis.c [hunk 2]
```diff
@@ -7858,17 +7858,17 @@ static void analysis_class_print(RzAnalysis *analysis, const char *class_name, b
 			RzAnalysisMethod *meth;
 			int i = 1;
 			rz_vector_foreach(methods, meth) {
-				RzList *row_list = rz_list_newf(free);
-				rz_list_append(row_list, rz_str_newf("%d", i++));
-				rz_list_append(row_list, rz_str_new(meth->real_name));
-				rz_list_append(row_list, rz_str_newf("0x%" PFMT64x, meth->addr));
+				RzPVector *row_vec = rz_pvector_new(free);
```
Comment (XVilka) [inline librz/core/cmd/cmd_analysis.c:7866]: NULL check

File: librz/util/table.c [hunk 3]
```diff
@@ -1288,24 +1335,17 @@ RZ_OWN RZ_API RzTable *rz_table_transpose(RZ_NONNULL RzTable *t) {
...
-				trans_row = trrow_iter->data;
-				rz_list_foreach (row->items, iter1, item) {
-					if (trrow_iter && trans_row->items) {
-						trans_row = trrow_iter->data;
-						rz_list_append(trans_row->items, strdup(item));
-						trrow_iter = trrow_iter->n;
-					}
-				}
+		rz_vector_foreach(t->rows, row) {
+			i = 0;
```
Comment (XVilka) [inline librz/util/table.c:1343]: Check if `row` is NULL before dereferencing.

File: librz/util/vector.c [hunk 4]
```diff
@@ -236,6 +236,50 @@ RZ_API void *rz_vector_flush(RzVector *vec) {
...
+	if (len <= 1) {
+		return;
+	}
+	size_t i = rand() % len, j = 0;
+	void *t, *pivot;
+
+	t = (void *)malloc(elem_size);
+	pivot = (void *)malloc(elem_size);
+	if (!t || !pivot) {
+		RZ_FREE(t);
```
Comment (XVilka) [inline librz/util/vector.c]: Use simple free, no need to set them to NULL since we immediately return afterwards.

File: librz/util/vector.c [hunk 5]
```diff
@@ -236,6 +236,50 @@ RZ_API void *rz_vector_flush(RzVector *vec) {
...
+	memcpy(VEC_INDEX(a, len - 1), VEC_INDEX(a, j), elem_size);
+	memcpy(VEC_INDEX(a, j), pivot, elem_size);
+	RZ_FREE(t);
+	RZ_FREE(pivot);
+	vector_quick_sort(a, elem_size, j, cmp, reverse);
+	vector_quick_sort(VEC_INDEX(a, j + 1), elem_size, len - j - 1, cmp, reverse);
+}
+#undef VEC_INDEX
+
+RZ_API void rz_vector_sort(RzVector *vec, RzVectorComparator cmp, bool reverse) {
```
Comment (XVilka) [inline librz/util/vector.c:285]: Add Doxygen.

**Review by XVilka — APPROVED** [review]
LGTM once those minor things are addressed.

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Pull request merged**